src/map.rs

```diff
@@ -151,10 +151,7 @@ impl<K: PartialEq, V, const N: usize> Map<K, V, N> {
     ///
     /// # Panics
     ///
-    /// It may panic if there are too many pairs in the map already. Pay attention,
-    /// it panics only in the "debug" mode. In the "release" mode, you are going to get
-    /// undefined behavior. This is done for the sake of performance, in order to
-    /// avoid a repetitive check for the boundary condition on every `insert()`.
+    /// It may panic if there are too many pairs in the map already.
```

Comment:
@FlareFlo now, since you modified this function, we must introduce another one, as you suggested: `insert_undefined`. Otherwise, we simply make this function slower without offering any alternative to the user.

Reply:
I'm fairly certain that these checks are negligible, unless benchmarks can show otherwise.

Reply:
@FlareFlo well, it's at least two CPU instructions. They can't be negligible :)

Reply:
> @FlareFlo well, it's at least two CPU instructions. They can't be negligible :)

The branch predictor and prefetcher will most likely disagree.

Reply:
@FlareFlo yes, most likely, but why should we take a chance. The ultimate objective of this library is to be _the fastest possible_. Not the most convenient, not the most user-friendly, but the fastest. Every CPU tick counts :)